### PR TITLE
Remove code deprecated in 2.5.0

### LINKED ISSUE #7069: Remove code deprecated in 2.5.0
This is a handle turning exercise - no coding skill is needed, you just need to be thorough :)

1. Find out what needs to be removed: `grep -R deprec zap/src/main/java | grep "2.5.0"` - there should be 41 matches
2. Remove the classes / methods / fields found above
3. Add a `ZAP` header to any `paros` code changed (see examples below)
4. Add the affected classes / methods / fields to `zap/zap.gradle.kts` (see below)
5. Run `spotlessApply`
6. Run `./gradlew japicmp` - it should pass, if not check the report it generates
7. Update the corresponding help file (see below)

Example PRs:

2.2.0 + 2.3.0: code: https://github.com/zaproxy/zaproxy/pull/7067 help: https://github.com/zaproxy/zap-core-help/pull/414
2.4.x: code: https://github.com/zaproxy/zaproxy/pull/7068 help: https://github.com/zaproxy/zap-core-help/pull/415

### PR DESCRIPTION
**Pull request by njmulsqb**
Signed-off-by: Najam Ul Saqib <najamulsaqib@tutamail.com>

Related to #7069 

Help has been updated in https://github.com/zaproxy/zap-core-help/pull/420

### CODE REVIEW

File: zap/zap.gradle.kts [hunk 1]
```diff
@@ -201,6 +201,7 @@ val japicmp by tasks.registering(JapicmpTask::class) {
         "org.parosproxy.paros.network.ConnectionParam#getProxyChainSkipName()",
         "org.parosproxy.paros.network.ConnectionParam#setProxyChainSkipName(java.lang.String)",
         "org.parosproxy.paros.view.AbstractFrame#loadIconImages()",
+        "org.parosproxy.paros.control.Control#createAndOpenUntitledDb()",
```
Comment (psiinon) [inline zap/zap.gradle.kts:204]: This should be in alphabetical order. Other than that this PR looks good so far :)

**Review by kingthorin — COMMENTED** [review]
Looks good to me

File: zap/zap.gradle.kts (hunk 1 above)
[the remark this answers is not in the document]
Comment (njmulsqb) [reply zap/zap.gradle.kts:204]: Doesn't pressing "resolve conversation" auto apply the suggestion? :P 
Do I need to make these changes locally and then push it?

[the remark this answers is not in the document]
Comment (psiinon) [reply zap/zap.gradle.kts:204]: Yeah, I just made a comment, I didnt actually make the changes :)

[the remark this answers is not in the document]
Comment (psiinon) [reply zap/zap.gradle.kts:204]: So yes, make the changes locally and push them

[the remark this answers is not in the document]
Comment (njmulsqb) [reply zap/zap.gradle.kts:204]: Oh Okay, I will do it

[the remark this answers is not in the document]
Comment (njmulsqb) [reply zap/zap.gradle.kts]: Done

**Comment by kingthorin** [comment]
Why do you keep merging?

**Comment by njmulsqb** [comment]
> Why do you keep merging?

I specifically didn't merge, I just took the pull from remote branch as otherwise it was not allowing me to push

**Comment by kingthorin** [comment]
Okay, we'll get this sorted out. Can you provide like a bullet summary of your workflow?

**Comment by njmulsqb** [comment]
> Okay, we'll get this sorted out. Can you provide like a bullet summary of your workflow?

Sure.
- I create a separate branch I am working on
- Create the changes on the branch
- Commit and sign off the changes
- Take latest pull to get the fresh code
- Push the code and make PR using Github UI

**Comment by kingthorin** [comment]
Okay skip this step "Take latest pull to get the fresh code"

If you need to update for some reason then fetch and rebase.

File: zap/src/main/java/org/parosproxy/paros/control/Control.java [hunk 2]
```diff
@@ -87,6 +87,7 @@
 // ZAP: 2021/09/13 Added setExitStatus.
 // ZAP: 2021/11/08 Validate if mandatory add-ons are present.
 // ZAP: 2022/02/09 No longer manage the proxy, deprecate related code.
+// ZAP: 2022/02/24 Remove code deprecated in 2.5.0
```
Comment (thc202) [inline zap/src/main/java/org/parosproxy/paros/control/Control.java:90]: I'd suggest using this sentence in all files, for consistency.

**Comment by thc202** [comment]
The commits should be fixed up.

**Comment by njmulsqb** [comment]
I think I messed up something, I took pull to get the commits I did here to make the `// ZAP` tag consistent and it brought other commits as well.
I rebased all of them and fixed into one.
Is it okay?

**Comment by njmulsqb** [comment]
> Okay skip this step "Take latest pull to get the fresh code"
> 
> If you need to update for some reason then fetch and rebase.

Sorry bud, I repeated the mistake I believe. :/

**Comment by psiinon** [comment]
Looks like you've pulled in an unrelated PR :/

**Comment by njmulsqb** [comment]
> Looks like you've pulled in an unrelated PR :/

How can I reverse this?

**Comment by thc202** [comment]
Do:
 - `git fetch upstream`
 - `git rebase upstream/main`

**Comment by njmulsqb** [comment]
> Do:
> 
>     * `git fetch upstream`
> 
>     * `git rebase upstream/main`

Done

**Comment by thc202** [comment]
You should have just commit then, you can (force) push to your remote.

**Comment by njmulsqb** [comment]
> You should have just one commit then, you can (force) push to your remote.

Wow, that worked like magic. I need to study the git commands you shared

**Comment by thc202** [comment]
Will fix the conflicts caused by the merge of other PR.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]
Nice :)

**Pull request merged**